applications/nrf5340_audio/src/bluetooth/bt_stream/unicast/unicast_client.c

```diff
@@ -889,38 +929,40 @@ static void discover_cb(struct bt_conn *conn, int err, enum bt_audio_dir dir)
 
 	for (int i = 0; i < CONFIG_CODEC_CAP_COUNT_MAX; i++) {
 		if (dir == BT_AUDIO_DIR_SINK && !err) {
-			memcpy(&unicast_servers[device_index].sink_codec_cap[i],
+			memcpy(&unicast_servers[lvl1_idx][lvl2_idx][lvl3_idx].sink_codec_cap[i],
 			       &temp_cap[temp_cap_index].codec[i],
 			       sizeof(struct bt_audio_codec_cap));
 		} else if (dir == BT_AUDIO_DIR_SOURCE && !err) {
-			memcpy(&unicast_servers[device_index].source_codec_cap[i],
+			memcpy(&unicast_servers[lvl1_idx][lvl2_idx][lvl3_idx].source_codec_cap[i],
 			       &temp_cap[temp_cap_index].codec[i],
 			       sizeof(struct bt_audio_codec_cap));
 		}
 	}
 
 	if (dir == BT_AUDIO_DIR_SINK && !err) {
-		if (valid_codec_cap_check(unicast_servers[device_index].sink_codec_cap,
-					  temp_cap[temp_cap_index].num_caps, BT_AUDIO_DIR_SINK,
-					  device_index)) {
-			bt_audio_codec_allocation_set(&lc3_preset_sink.codec_cfg,
-						      unicast_servers[device_index].location);
+		if (valid_codec_cap_check(
+			    unicast_servers[lvl1_idx][lvl2_idx][lvl3_idx].sink_codec_cap,
+			    temp_cap[temp_cap_index].num_caps, BT_AUDIO_DIR_SINK, lvl3_idx)) {
```

Comment:
Would appreciate an explanation here.

Reply:
The only thing that changes here is the use of the three level index. What happens is that it checks if the capabilities of the unicast_server matches what the unicast_client can support. If it does we store the location, if it doesn't we remove the endpoint.